### ISSUE TITLE
Sign helm chart and provide provenance

### ISSUE THREAD

**Issue by PrivatePuffin** [minio/operator#1855]
**Is your feature request related to a problem? Please describe.**
With supply chain attacks on the rise, it's important that all parts of the supply chain are correctly signed and secured.

**Describe the solution you'd like**
Hence it's best practice for professional software to sign their Helm charts as well and provide provenance

**Additional context**
Instructions available here:
https://helm.sh/docs/topics/provenance/

Example available here:
https://github.com/truecharts/helm-staging/blob/main/.github/workflows/release.yaml

**Comment by PrivatePuffin** [comment]
not stale

**Comment by ramondeklein** [comment]
I don't think we will sign our Helm charts. It will complicate our release process, because we only have limited people that have access to the signing keys.

**Comment by PrivatePuffin** [comment]
A shame to see that security and traceability not a priority are not a priority for Minio.

Considering the minor inconvenience of adding a literall 5 lines to a release pipeline, takes precedence over security and traceability, imho disqualifies Minio for any enterprise-level use case and also means imho, it shouldn't pass any reasonably-strict security certification.

I'm very sorry to hear this, I saw a bright future for wider adoption of Minio.
But this definitely crushes that opinion.

**Comment by ramondeklein** [comment]
MinIO already has a wide adoption and takes security seriously. Besides your request, we haven't had any request for signed Helm charts and this issue didn't get any upvotes. The problem isn't the signing process itself, it's the management of code-signing keys. Only a few people within MinIO have access to code signing keys and this would mean that these people need to be involved with signing the Helm charts. Because there is no request for this feature, we don't implement it. We need to prioritize what we do and don't do.
